pyccel/codegen/printing/fcode.py

```diff
@@ -938,8 +952,12 @@ def _print_Declare(self, expr):
             (not(allocatable or is_pointer) or is_static or is_stack_array)):
             #TODO fix bug when we inclue shape of type list
 
-            rankstr =  ','.join('{0}:{1}-1'.format(self._print(s),
-                                                 self._print(i)) for i in shape)
+            if var.order=='C':
+                rankstr =  ','.join('{0}:{1}-1'.format(self._print(s),
+                                                    self._print(i)) for i in shape[::-1])
+            else:
+                rankstr =  ','.join('{0}:{1}-1'.format(self._print(s),
+                                                     self._print(i)) for i in shape)
```

Comment:
> Why are we not reversing the shape anymore?

@yguclu  This is due to these lines. The shape was printed in the same way for C-ordered and F-ordered arrays. I'm not sure how we got away with this before. I think it was due to the f2py line and the fact that all the initialisation was reversed anyway for arrays. Once this has been corrected in order to correctly print the tuples, the f2py sizes no longer match . However this is corrected by the comment.

I think it would be worth looking into this a bit more to check that I haven't broken anything. For the moment I am relying on the fact that the tests pass to ensure that the logic is correct, but I don't think I have fully understood why it was working before. @saidctb do you understand this?

Reply:
Actually I played around with various solutions before finding the one that was implemented in the f2py printer before. I also discussed with Said a lot about it.

@EmilyBourne You might have found a different way to achieve the same goal. Interesting..

Reply:
I wonder if we even need to have an f2py comment for C ordering now..

Reply:
yes maybe we don't need the comments now

Reply:
Apparently we really need them, otherwise some tests fail.